math/src/main/java/com/powsybl/math/matrix/ComplexMatrix.java

```diff
@@ -0,0 +1,127 @@
+                transposed.imagPartMatrix.set(i, j, imagPartMatrix.get(j, i));
+            }
+        }
+        return transposed;
+    }
+
+    public ComplexMatrix scale(Complex factor) {
+        ComplexMatrix scaled = new ComplexMatrix(getRowCount(), getColumnCount());
+        for (int i = 0; i < getRowCount(); i++) {
+            for (int j = 0; j < getColumnCount(); j++) {
+                scaled.realPartMatrix.set(i, j, realPartMatrix.get(i, j) * factor.getReal() - imagPartMatrix.get(i, j) * factor.getImaginary());
+                scaled.imagPartMatrix.set(i, j, imagPartMatrix.get(i, j) * factor.getReal() + realPartMatrix.get(i, j) * factor.getImaginary());
+            }
+        }
+        return scaled;
+    }
+
+    public ComplexMatrix scale(double factor) {
+        return scale(new Complex(factor, 0.));
+    }
+
+    // utils to switch between complex and real cartesian representation of a complex matrix
+    public DenseMatrix toRealCartesianMatrix() {
+        DenseMatrix realMatrix = new DenseMatrix(getRowCount() * 2, getColumnCount() * 2);
+        for (int i = 0; i < getRowCount(); i++) {
+            for (int j = 0; j < getColumnCount(); j++) {
+                Complex complexTerm = new Complex(realPartMatrix.get(i, j), imagPartMatrix.get(i, j));
+                realMatrix.add(2 * i, 2 * j, complexTerm.getReal());
+                realMatrix.add(2 * i + 1, 2 * j + 1, complexTerm.getReal());
+                realMatrix.add(2 * i, 2 * j + 1, -complexTerm.getImaginary());
+                realMatrix.add(2 * i + 1, 2 * j, complexTerm.getImaginary());
+            }
+        }
+
+        return realMatrix;
+    }
+
+    public static ComplexMatrix fromRealCartesian(DenseMatrix realMatrix) {
+        int columnCount = realMatrix.getColumnCount();
+        int rowCount = realMatrix.getRowCount();
+        if (columnCount % 2 != 0 || rowCount % 2 != 0) { // dimensions have to be even
+            throw new MatrixException("Incompatible matrices dimensions to build a complex matrix from a real cartesian");
+        }
+
+        ComplexMatrix complexMatrix = new ComplexMatrix(rowCount / 2, columnCount / 2);
+        for (int i = 0; i < rowCount / 2; i++) {
+            for (int j = 0; j < columnCount / 2; j++) {
+
+                int rowIndexInCartesian = 2 * i;
+                int colIndexInCartesian = 2 * j;
+
+                // Before building the complex matrix term, check that the 4x4 cartesian bloc can be transformed into a Complex term
+                double t11 = realMatrix.get(rowIndexInCartesian, colIndexInCartesian);
+                double t12 = realMatrix.get(rowIndexInCartesian, colIndexInCartesian + 1);
+                double t21 = realMatrix.get(rowIndexInCartesian + 1, colIndexInCartesian);
+                double t22 = realMatrix.get(rowIndexInCartesian + 1, colIndexInCartesian + 1);
+
+                double epsilon = 0.00000001;
+                if (FastMath.abs(t11 - t22) > epsilon) {
```

Comment:
These two if bloc can be shared, and with the first exception of your method, you can introduce a string variable for exception message.

Reply:
Done